packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2249,20 +2248,20 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
   static bool debugCheckingIntrinsics = false;
 
   bool _debugRelayoutBoundaryAlreadyMarkedNeedsLayout() {
-    if (_relayoutBoundary == null) {
+    if (_isRelayoutBoundary == null) {
       // We don't know where our relayout boundary is yet.
       return true;
     }
     RenderObject node = this;
-    while (node != _relayoutBoundary) {
-      assert(node._relayoutBoundary == _relayoutBoundary);
+    while (node._isRelayoutBoundary != true) {
+      assert(node._isRelayoutBoundary == false);
       assert(node.parent != null);
       node = node.parent!;
       if ((!node._needsLayout) && (!node._debugDoingThisLayout)) {
         return false;
       }
     }
-    assert(node._relayoutBoundary == node);
+    assert(node._isRelayoutBoundary ?? false);
```

Comment:
For readibility and understandability, I'd find it clearer to express this assert as

```suggestion
    assert(node._isRelayoutBoundary == true);
```

to indicate that that node absolutely must be a relayout boundary.

Reply:
Oh, the `use_if_null_to_convert_nulls_to_bools` lint probably forced this? Never mind then.